### PR TITLE
ROX-20886: Add modal for viewing exception request comments

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds the ability to view comments on the Request Details page. Clicking the comments button/link will open a modal to view comments in a scrollable manner.

## Screenshots

Screenshot 2023-11-15 at 10 23 07 AM
Screenshot 2023-11-15 at 10 23 10 AM

### CODE REVIEW

**Comment by sachaudh** [comment]
Also, discovered you can approve a request multiple times using the API. We won't allow it on the UI, when I add the approval flow, but just raising awareness.

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx [hunk 1]
```diff
@@ -155,3 +167,49 @@ export function RequestedItems({ exception, context }: RequestedItemsProps) {
...
+        <>
+            <Button variant="link" isInline onClick={openModal}>{`${comments.length} ${pluralize(
+                'comment',
+                comments.length
+            )}`}</Button>
+            <Modal variant="small" title="Comments" isOpen={isModalOpen} onClose={closeModal}>
+                <List isPlain isBordered>
+                    {comments.map((comment) => {
+                        return (
+                            <ListItem>
```
Comment (dvail) [inline ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:206]:
```suggestion
                            <ListItem key={comment.id}>
```

File: ui/apps/platform/src/services/VulnerabilityExceptionService.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ export type VulnerabilityState = 'OBSERVED' | 'DEFERRED' | 'FALSE_POSITIVE';
 
 export type ExceptionStatus = 'PENDING' | 'APPROVED' | 'DENIED' | 'APPROVED_PENDING_UPDATE';
 
-type Comment = {
+export type VulnerabilityExceptionComment = {
```
Comment (dvail) [inline ui/apps/platform/src/services/VulnerabilityExceptionService.ts:15]: Good idea on the more descriptive naming for export 👍

**Review by dvail — APPROVED** [review]
LGTM - one suggestion to silence a React console error

File: ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx (hunk 1 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/Vulnerabilities/ExceptionManagement/components/ExceptionRequestTableCells.tsx:206]: Ah, I keep forgetting about this one 🤦🏼‍♂️

**Pull request merged**